### PR TITLE
Disable drum roll when volume is muted

### PR DESCRIPTION
**Pull request by mohammedhemaid**
The video of the feature:
https://photos.app.goo.gl/exDiabuoiput5GuQA

### CODE REVIEW

File: app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java [hunk 1]
```diff
@@ -154,8 +164,12 @@ private void playSound() {
             UIUtils.showLongToast(R.string.drumroll_error, this);
         }
 
+        handleNamesAnimations(2600);
+    }
+
+    private void handleNamesAnimations(int delay) {
```
Comment (mohammedhemaid) [inline app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java]: should I change this to delayNamesAnimations?

File: app/app.iml [hunk 2]
```diff
@@ -30,15 +30,15 @@
...
-      <sourceFolder url="file://$MODULE_DIR$/build/generated/renderscript_source_output_dir/debug/out" isTestSource="false" generated="true" />
       <sourceFolder url="file://$MODULE_DIR$/build/generated/res/rs/debug" type="java-resource" generated="true" />
-      <sourceFolder url="file://$MODULE_DIR$/build/generated/ap_generated_sources/debugAndroidTest/out" isTestSource="true" generated="true" />
       <sourceFolder url="file://$MODULE_DIR$/build/generated/aidl_source_output_dir/debugAndroidTest/out" isTestSource="true" generated="true" />
-      <sourceFolder url="file://$MODULE_DIR$/build/generated/source/buildConfig/androidTest/debug" isTestSource="true" generated="true" />
       <sourceFolder url="file://$MODULE_DIR$/build/generated/renderscript_source_output_dir/debugAndroidTest/out" isTestSource="true" generated="true" />
+      <sourceFolder url="file://$MODULE_DIR$/build/generated/ap_generated_sources/debugAndroidTest/out" isTestSource="true" generated="true" />
+      <sourceFolder url="file://$MODULE_DIR$/build/generated/source/buildConfig/androidTest/debug" isTestSource="true" generated="true" />
```
Comment (mohammedhemaid) [inline app/app.iml:41]: I believe I shouldn't track some file, I accidentally added it

File: app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java [hunk 3]
```diff
@@ -134,12 +136,20 @@ private void chooseNames() {
             names.setAlpha(0.0f);
             names.setText(chosenNamesText);
 
-            playSound();
+            if (isMediaVolumeMuted()) {
+                handleNamesAnimations(300);
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java]: Why do we need 300ms delay when it's muted? I feel like the names should just animate in ASAP if there's no volume? Also, how did we get 300?

File: app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java [hunk 4]
```diff
@@ -134,12 +136,20 @@ private void chooseNames() {
...
+            } else {
+                playSound();
+            }
         } else {
             UIUtils.showLongToast(R.string.no_names_left, this);
         }
     }
 
+    private boolean isMediaVolumeMuted() {
+        return ((AudioManager) getSystemService(Context.AUDIO_SERVICE)).getStreamVolume(AudioManager.STREAM_MUSIC) == 0;
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java]: I don't think this is worth the method. Let's give it back to the dex.

File: app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java (hunk 1 above)
Comment (Gear61) [reply app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java]: I think playNamesAnimation(int delayMs) is better. More descriptive variable and matches more of what the method does IMHO.

**Review by Gear61 — CHANGES_REQUESTED** [review]

File: app/app.iml (hunk 2 above)
Comment (Gear61) [reply app/app.iml:41]: I'm gonna remove this from the repo soon. It's code-gen.

File: app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java [hunk 5]
```diff
@@ -134,7 +136,11 @@ private void chooseNames() {
             names.setAlpha(0.0f);
             names.setText(chosenNamesText);
 
-            playSound();
+            if (((AudioManager) getSystemService(Context.AUDIO_SERVICE)).getStreamVolume(AudioManager.STREAM_MUSIC) == 0) {
+                playNamesAnimation(getResources().getInteger(R.integer.default_anim_length));
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationActivity.java:140]: I feel like @BindInt is cleaner.

**Review by Gear61 — APPROVED** [review]
Have some minor comments but this looks good to me. I'll merge and polish.

**Pull request merged**